Replace custom hero markup with DS hero component for Article & Methodology pages (CMS-307)

janehughes opened the pull request:
### What is the context of this PR?

This change updates the statistical article page & methodology pages to use the latest hero component grey variant from the design system.

### How to review

- Pull branch locally
- Preview a statistical article page type
- See that the hero looks and behaves as expected
- Check the template and see that we are now using the design system hero component and the custom code has been removed

<details>
  <summary>Local preview</summary>

![statistical-article-hero](https://github.com/user-attachments/assets/8d0fc565-7a27-4397-92fa-c548a02df4e9)
</details>


### Follow-up Actions

- Since removing the custom hero markup, some spacing issues appeared above the headline figures which will need addressing as part of headline figures updates (currently with DS team)
- Further design updates for the hero are in progress with DS team so this may need updating again later on

helenb reviewed (commented):
Thanks @janehughes - I have reviewed and tested the front-end side of this, and added some suggestions regarding the rich text.

Can we also apply this header to other content types? If so, we may be able to remove the custom CSS for the headers as well.

janehughes commented on cms/jinja2/templates/pages/statistical_article_page.html:
```diff
@@ -1,66 +1,95 @@
-                    <h1 class="ons-u-fs-3xl common-header__heading">
-                        {{ page.display_title }}
-                    </h1>
+    {# Currently missing email link as the onsDescriptionList does not allow markup #}
```
Thank you Helen! Updated

janehughes commented on cms/jinja2/templates/pages/statistical_article_page.html:
```diff
@@ -1,66 +1,95 @@
+    {% else %}
+        {% set contact_details = "" %}
+    {% endif %}
+    {# Currently missing links as the onsDescriptionList does not allow markup #}
```
Updated this too 👍

janehughes commented:
> Thanks @janehughes - I have reviewed and tested the front-end side of this, and added some suggestions regarding the rich text.
> 
> Can we also apply this header to other content types? If so, we may be able to remove the custom CSS for the headers as well.

I've updated the methodology page as well and updated the PR title/desc to include both stat article & methodology page. I've just done these for now as they're both using the grey variant so it keeps things consistent. I'm not sure if any other pages are ready to be updated yet.

zerolab commented on cms/jinja2/templates/pages/methodology_page.html:
```diff
@@ -2,48 +2,82 @@
+                    contact_details,
+                ]
+            },
+        "breadcrumbs":
+            {
+            "ariaLabel": _("Breadcrumbs"),
+            "itemsList": breadcrumbs(page),
+            }
```
```suggestion
            "breadcrumbs": {
                "ariaLabel": _("Breadcrumbs"),
                "itemsList": breadcrumbs(page),
            }
```

zerolab commented on cms/jinja2/templates/pages/statistical_article_page.html:
```diff
@@ -2,74 +2,110 @@
+                    contact_details,
+                ]
+            },
+        "breadcrumbs":
+            {
+            "ariaLabel": _("Breadcrumbs"),
+            "itemsList": breadcrumbs(page),
+            }
+        })
```
```suggestion
            "breadcrumbs": {
                "ariaLabel": _("Breadcrumbs"),
                "itemsList": breadcrumbs(page),
            }
        })
```

janehughes replied:
Thank you! Updated

zerolab reviewed (approved):
Just a couple of formatting suggestions from me

sanjeevz3009 commented on cms/navigation/templatetags/navigation_tags.py:
```diff
@@ -121,3 +125,17 @@ def footer_menu_columns(
+    breadcrumbs_list = []
+    request = context.get("request")
+    for ancestor_page in page.get_ancestors().specific().defer_streamfields():
+        if not ancestor_page.is_root():
```
Nit: 

```
# Skip the root page entirely
if ancestor_page.is_root():
    continue

if ancestor_page.depth <= BREADCRUMBS_HOMEPAGE_DEPTH:
    breadcrumb_items.append({"url": "/", "text": _("Home")})
elif not getattr(ancestor_page, "exclude_from_breadcrumbs", False):
    breadcrumb_items.append(
        {
            "url": ancestor_page.get_url(request=request),
            "text": ancestor_page.title,
        }
    )
```

De-nests an `if`

MaciekBaron replied:
Addressed in [0c5b3c6](https://github.com/ONSdigital/dis-wagtail/pull/129/commits/0c5b3c69c808984abc218ed51e6447ec2d2db893)

sanjeevz3009 reviewed (approved):
Minor comment, other than looks good =)

MebinAbraham commented on cms/jinja2/templates/pages/statistical_article_page.html:
```diff
@@ -2,74 +2,109 @@
+            "officialStatisticsBadge": page.is_accredited,
+            "wide": true,
+            "detailsColumns": 12,
+            "censusLogo": page.is_census,
```
This doesn't seem to be working as there are no changes to the hero, when is census is ticked. Does this fall under the follow up action point of `Further design updates for the hero are in progress with DS team so this may need updating again later on`, or a bug in DS?

helenb replied:
It also worked fine for me when I tested.

MebinAbraham replied:
Had marked it as resolved after I posted. User error, I didn't load the latest DS templates.

MebinAbraham reviewed (approved):
Would like to look at some deduplication in future, as there are shared elements that we can split to partials.

MebinAbraham reviewed (changes requested):
Could we add some simple functional tests to ensure hero is as expected. i.e When census is ticked, the census logo is visible etc. We want to have confidence that we have configured the hero correctly and any breaking changes to DS are caught by our tests.

helenb reviewed (approved):
Looking good.

MaciekBaron commented:
@MebinAbraham additional integration tests added in [af02a46](https://github.com/ONSdigital/dis-wagtail/pull/129/commits/af02a469920d718dd4f65c9a2c30f42e07da98e8)
